bin/main

```diff
@@ -1,2 +1,96 @@
 #!/usr/bin/env ruby
-puts 'Hello world!'
+require_relative '../lib/game_logic'
+
+class Player
+  attr_accessor :name
+
+  def initialize(name)
+    @name = name
+  end
+end
+
+def greeting
+  puts 'TIC TAC TOE GAME'
+  puts 'TO START PRESS ENTER'
+  gets.chomp
+end
+
+puts greeting
+
+puts 'Please enter Player 1 name'
+player1 = Player.new(gets.chomp)
+puts "\nPlease enter Player 2 name"
+player2 = Player.new(gets.chomp)
+puts "\n#{player1.name.capitalize} will play as X and #{player2.name.capitalize} will play as O"
+sleep 1
+puts "\nLet's play!"
+
+slots = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+def design_board(slots)
+  system('clear')
+  system('cls')
+
+  puts '+---+---+---+'
+  puts "| #{slots[0]} | #{slots[1]} | #{slots[2]} |"
+  puts '+---+---+---+'
+  puts "| #{slots[3]} | #{slots[4]} | #{slots[5]} |"
+  puts '+---+---+---+'
+  puts "| #{slots[6]} | #{slots[7]} | #{slots[8]} |"
+  puts '+---+---+---+'
+end
+
+# winner = false
+tie = false
+
+def player1_turn(arr, name, remaining_slots)
+  puts "#{name} is your turn to play, choose one of the remaining slots #{remaining_slots}"
+  choice = gets.chomp.to_i
+  arr.map! do |item|
+    if item == choice
+      arr[arr.index(item)] = 'X'
+    else
+      item
+    end
+  end
+end
+
+def player2_turn(arr, name, remaining_slots)
+  puts "#{name} is your turn to play, choose one of the remaining slots #{remaining_slots}"
+  choice = gets.chomp.to_i
+  arr.map! do |item|
+    if item == choice
+      arr[arr.index(item)] = 'O'
+    else
+      item
+    end
+  end
+end
```

Comment:
These two functions are doing the same job. Please make only one. Remember the DRY (Don't Repeat Yourself) rule.